pkg/shell/util.tsx

```diff
@@ -172,8 +183,11 @@ export class CompiledComponents {
         // Still don't know where it comes from, check for parent
         if (!component) {
             const comp = this.manifests[path];
-            if (comp && comp.parent && comp.parent.component)
-                component = comp.parent.component as string;
+            if (comp && comp.parent) {
+                const parent = comp.parent as ManifestParentSection;
+                if (parent.component)
+                    component = parent.component;
+            }
```

Comment:
So you are asking for "The Shell should validate the schema of all manifests". That's a good idea. We could have done that without TypeScript all these years, but we didn't.  A bug in a manifests is like a bug in JavaScript, and should be found and fixed by the developer.

And while TypeScript might benefit from it once schema validation is done (think generating TS types from JSON schemas), I don't think TypeScript by itself is gonna help. All it does it point out our sins...

So, can I ask to keep the manifest schema validation yak out of this PR? :-)

Reply:
Summary from meeting: My gut feeling/slight preference is the `get_json_string()` kind of accessor on usage, but I agree that for common keys we may rather want some more centralized "schema validation" thing, with a constructor. My minimum requirement here would be to add a `// HACK: unsafe "as"` or similar so that we can find these again.

Reply:
There is now `import_Manifests` and `import_ShellManifest` that concentrate this issue in one place.

I still want to fully type our Manifest data structures and then access them from our code without any additional accessors. I.e., I want to do this:

```
interface Foo {
  name: string;
}

function bar(f: Foo) {
  console.log("Hi", f.name.toUpper());
}

bar(import_Foo(get_some_json());
```

and not

```
function bar(f: JsonValue) {
  console.log("Hi", json_as_string(json_get(f, "name"), "<unknown>").toUpper());
}

bar(get_some_json());
```

This needs a `import_Foo` function, which is not as trivial to write as the accessors, but I still want it for Christmas! :-)